src/Modelling/PetriNet/Reach/Reach.hs

```diff
@@ -478,21 +553,18 @@ defaultReachInstance = ReachInstance {
   withMinLengthHint = False
 }
 
-generateNetGoal
-  :: (MonadCatch m, MonadDiagrams m, MonadGraphviz m)
+possibleNetGoals
+  :: MonadRandom m
   => NetGoalConfig
-  -> Int
-  -> m (NetGoal Place Transition)
-generateNetGoal NetGoalConfig {..} seed = do
+  -> m [(Net Place Transition, State Place)]
+possibleNetGoals NetGoalConfig {..} =
   let ps = [Place 1 .. Place numPlaces]
       tries = forM [1 :: Int .. 1000] $ const $ do
         n <- netLimits vLow vHigh nLow nHigh
             ps
             ts
             capacity
         return $ do
-          -- Filter out nets with isolated nodes
-          guard $ not $ hasIsolatedNodes n
```

Comment:
@marcellussiegburg, same issue here.

I see that the following commit did something about `hasIsolatedNodes`: https://github.com/fmidue/modelling-tasks/commit/6ba60297804f543c43d243314621a0f89838fc9d

But I fail to see how that commit was just a refactoring, i.e., not changing semantics.

It moved guards out of functions that generate their nets with the `netLimits` function, and in exchange added conditions elsewhere, but that elsewhere is **not** affecting `netLimits` at all, it seems.